From 01b7b10d0b3002a5ef55891467e9f174d4fea9e3 Mon Sep 17 00:00:00 2001
From: Pino Toscano <toscano.pino@tiscali.it>
Date: Mon, 16 May 2011 07:07:57 +0000
Subject: shm: Don't use PATH_MAX

PATH_MAX is not defined on GNU Hurd and others. Also fix
format string.
---
diff --git a/sys/shm/shmpipe.c b/sys/shm/shmpipe.c
index 6465e38..38711f9 100644
--- a/sys/shm/shmpipe.c
+++ b/sys/shm/shmpipe.c
@@ -262,7 +262,7 @@ static ShmArea *
 sp_open_shm (char *path, int id, mode_t perms, size_t size)
 {
   ShmArea *area = spalloc_new (ShmArea);
-  char tmppath[PATH_MAX];
+  char tmppath[32];
   int flags;
   int prot;
   int i = 0;
@@ -285,7 +285,7 @@ sp_open_shm (char *path, int id, mode_t perms, size_t size)
     area->shm_fd = shm_open (path, flags, perms);
   } else {
     do {
-      snprintf (tmppath, PATH_MAX, "/shmpipe.5%d.%5d", getpid (), i++);
+      snprintf (tmppath, sizeof (tmppath), "/shmpipe.%5d.%5d", getpid (), i++);
       area->shm_fd = shm_open (tmppath, flags, perms);
     } while (area->shm_fd < 0 && errno == EEXIST);
   }
--
cgit v0.8.3-6-g21f6
